source/puppet/5.3/hiera_migrate.md

```diff
@@ -161,121 +164,123 @@ Consider this example `hiera.yaml` version 3 file:
+
+6.  Remove hierarchy levels that use `calling_module`, `calling_class`, and `calling_class_path`, which were allowed pseudo-variables in Hiera 3.
+
+    Hiera.yaml version 5 does not support these. Remove hierarchy levels that interpolate them.
+
+    Anything you were doing with these variables is better accomplished by using the module data layer, or by using the glob pattern (if the reason for using them was to enable splitting up data into multiple files, and not knowing in advance what they names of those would be)
+
+7.  Translate built-in backends to the version 5 config, where the hierarchy is written as an array of hashes. For hierarchy levels that use the built-in backends, for example YAML and JSON, use the `data_hash` key to set the backend. See Configuring a hierarchy level in the `hiera.yaml` v5 reference for more information.
+
+    Set the following keys:
+
+    -   `name` — A human-readable name.
+    -   `path` or `paths` — The path you used in your version 3 `hiera.yaml` hierarchy, but with a file extension appended.
+    -   `data_hash` — The backend to use `yaml_data` for YAML, `json_data` for JSON.
+    -   `datadir` — The data directory. In version 5, it's relative to the `hiera.yaml` file's directory.
+
+    If you have set default values for `data_hash` and `datadir`, you can omit them.
+
+    ``` yaml
+    version: 5
+    defaults:
+      datadir: data
+      data_hash: yaml_data
+    hierarchy:
+      - name: "Per-node data (yaml version)"
+        path: "nodes/%{trusted.certname}.yaml" # Add file extension.
+        # Omitting datadir and data_hash to use defaults.
+
+      - name: "Other YAML hierarchy levels"
+        paths: # Can specify an array of paths instead of one.
+          - "location/%{facts.whereami}/%{facts.group}.yaml"
+          - "groups/%{facts.group}.yaml"
+          - "os/%{facts.os.family}.yaml"
+          - "common.yaml"
     ```
-      
- 8. Translate hiera-eyaml backends, which work in a similar way to the other built-in backends. The differences are:
- 
-   The `hiera-eyaml` gem has to be installed, and you need a key-pair.
+
+8.  Translate hiera-eyaml backends, which work in a similar way to the other built-in backends. The differences are:
+
+    The `hiera-eyaml` gem has to be installed, and you need a key-pair.
 a different backend setting. Instead of `data_hash: yaml`, use `lookup_key: eyaml_lookup_key`.
 Each hierarchy level needs an `options` key with paths to the public and private keys. You cannot set a global default for this.
 
-   ```
+    ``` yaml
     - name: "Per-group secrets"
        path: "groups/%{facts.group}.eyaml"
        lookup_key: eyaml_lookup_key
        options:
          pkcs7_private_key: /etc/puppetlabs/puppet/eyaml/private_key.pkcs7.pem
          pkcs7_public_key:  /etc/puppetlabs/puppet/eyaml/public_key.pkcs7.pem
-   ```
-      
-9. Translate custom Hiera 3 backends
```

Comment:
The ordered list broke down here because the code fences in the block before it weren't consistently indented.